src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,49 +1,103 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  currntUserId: number | null,
+  selectedTodo: Todo | null,
+  onTodoInfoRequestedChange: (arg: boolean) => void,
+}
+
+export const TodoModal: React.FC<Props> = ({
+  currntUserId,
+  selectedTodo,
+  onTodoInfoRequestedChange,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [userUploaded, setUserUploaded] = useState(false);
+  const [isUSerError, setIsUSerError] = useState(false);
+
+  useEffect(() => {
+    const loadUser = async (id: number) => {
+      try {
+        const loadedUser = await getUser(id);
+
+        setUser(loadedUser);
+        setUserUploaded(true);
+      } catch (error) {
+        setUserUploaded(true);
+        setIsUSerError(true);
+      }
+    };
+
+    if (currntUserId) {
+      loadUser(currntUserId);
+    }
```

Comment:
If you render modal only when you have selected todo, you always will have `userId`

```suggestion
    loadUser(selectedTodo.userId);
```